src/tile.cpp

```diff
@@ -236,19 +233,19 @@ const Creature* Tile::getBottomCreature() const
 	return nullptr;
 }
 
-Creature* Tile::getTopVisibleCreature(const Creature* creature) const
+std::shared_ptr<Creature> Tile::getTopVisibleCreature(std::shared_ptr<const Creature> creature) const
 {
 	if (const CreatureVector* creatures = getCreatures()) {
 		if (creature) {
-			for (Creature* tileCreature : *creatures) {
+			for (const auto& tileCreature : *creatures) {
 				if (creature->canSeeCreature(tileCreature)) {
 					return tileCreature;
 				}
 			}
 		} else {
-			for (Creature* tileCreature : *creatures) {
+			for (const auto& tileCreature : *creatures) {
 				if (!tileCreature->isInvisible()) {
-					const Player* player = tileCreature->getPlayer();
+					auto player = tileCreature->getPlayer();
```

Comment:
Could this be `const auto`
```suggestion
					const auto player = tileCreature->getPlayer();
```